File: solutions/javascript/rna-transcription/1/rna-transcription.js
//
// This is only a SKELETON file for the "Leap" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var DnaTranscriber = function () {
	//this.dna = dna;
};

DnaTranscriber.prototype.toRna = function (dna) {
var rna =""
for (var i = 0; i<dna.length; i++) {
	switch(dna.substring(i,i+1)) {
	  case "G": 
	    rna = rna + "C";
	    break;
	  case "C": 
	    rna = rna + "G";
	    break;
	  case "T": 
	    rna = rna + "A";
	    break;
	  case "A": 
	    rna = rna + "U";
	   break;
	   default:
		throw new Error('Invalid input');
	}	
}	

return rna;

};

module.exports = DnaTranscriber;

File: solutions/r/rna-transcription/1/rna-transcription.R
to_rna <- function(dna) {
	rna <- ""
	for (d in unlist(strsplit(dna, ""))) {
		if (d == "G") {r <- "C"}
		else if (d == "C") {r <- "G"}
		else if (d == "T") {r <- "A"}
		else if (d == "A") {r <- "U"}
		else { stop(error) }
		rna <- paste(rna, r, sep="")
		#print (d)
		#print (rna)
	}
	return (rna)
}
